File: electron/src/renderer/lib/api/minutes_template.ts
/**
 * Minutes Template API Client
 */
import api from '../apiClient';

export interface MinutesTemplate {
  id: string;
  name: string;
  code?: string;
  description?: string;
  structure: any; // JSONB structure
  sample_data?: any;
  meeting_types?: string[];
  is_default: boolean;
  is_active: boolean;
  version: number;
  created_by?: string;
  created_at?: string;
  updated_by?: string;
  updated_at?: string;
  parent_template_id?: string;
}

export interface MinutesTemplateList {
  templates: MinutesTemplate[];
  total: number;
}

export interface MinutesTemplateCreate {
  name: string;
  code?: string;
  description?: string;
  structure: any;
  sample_data?: any;
  meeting_types?: string[];
  is_default?: boolean;
  is_active?: boolean;
  created_by?: string;
}

export interface MinutesTemplateUpdate {
  name?: string;
  code?: string;
  description?: string;
  structure?: any;
  sample_data?: any;
  meeting_types?: string[];
  is_default?: boolean;
  is_active?: boolean;
  updated_by?: string;
}

const ENDPOINT = '/minutes-templates';

export const minutesTemplateApi = {
  /**
   * List all templates
   */
  list: async (params?: {
    meeting_type?: string;
    is_active?: boolean;
    skip?: number;
    limit?: number;
  }): Promise<MinutesTemplateList> => {
    return api.get<MinutesTemplateList>(ENDPOINT, params);
  },

  /**
   * Get default template
   */
  getDefault: async (): Promise<MinutesTemplate> => {
    return api.get<MinutesTemplate>(`${ENDPOINT}/default`);
  },

  /**
   * Get template by ID
   */
  get: async (templateId: string): Promise<MinutesTemplate> => {
    return api.get<MinutesTemplate>(`${ENDPOINT}/${templateId}`);
  },

  /**
   * Create template
   */
  create: async (data: MinutesTemplateCreate): Promise<MinutesTemplate> => {
    return api.post<MinutesTemplate>(ENDPOINT, data);
  },

  /**
   * Update template
   */
  update: async (templateId: string, data: MinutesTemplateUpdate): Promise<MinutesTemplate> => {
    return api.put<MinutesTemplate>(`${ENDPOINT}/${templateId}`, data);
  },

  /**
   * Delete template
   */
  delete: async (templateId: string): Promise<void> => {
    return api.delete<void>(`${ENDPOINT}/${templateId}`);
  },

  /**
   * Set template as default
   */
  setDefault: async (templateId: string): Promise<MinutesTemplate> => {
    return api.post<MinutesTemplate>(`${ENDPOINT}/${templateId}/set-default`, {});
  },
};

export default minutesTemplateApi;

File: backend/app/schemas/minutes_template.py
"""
Minutes Template Schemas
"""
from pydantic import BaseModel, Field
from typing import Optional, List, Dict, Any
from datetime import datetime
from uuid import UUID


class TemplateField(BaseModel):
    """Template field structure"""
    id: str
    label: str
    type: str  # text, datetime, array, signature, etc.
    required: bool = False
    source: str  # meeting.title, meeting.participants, ai_generated
    structure: Optional[Dict[str, Any]] = None  # For array/object types


class TemplateSection(BaseModel):
    """Template section structure"""
    id: str
    title: str
    order: int
    required: bool = True
    fields: List[TemplateField]


class TemplateStructure(BaseModel):
    """Template structure"""
    sections: List[TemplateSection]
    formatting: Optional[Dict[str, Any]] = None


class MinutesTemplateBase(BaseModel):
    name: str
    code: Optional[str] = None
    description: Optional[str] = None
    structure: Dict[str, Any]  # JSONB structure
    sample_data: Optional[Dict[str, Any]] = None
    meeting_types: Optional[List[str]] = None
    is_default: bool = False
    is_active: bool = True


class MinutesTemplateCreate(MinutesTemplateBase):
    created_by: Optional[str] = None


class MinutesTemplateUpdate(BaseModel):
    name: Optional[str] = None
    code: Optional[str] = None
    description: Optional[str] = None
    structure: Optional[Dict[str, Any]] = None
    sample_data: Optional[Dict[str, Any]] = None
    meeting_types: Optional[List[str]] = None
    is_default: Optional[bool] = None
    is_active: Optional[bool] = None
    updated_by: Optional[str] = None


class MinutesTemplateResponse(MinutesTemplateBase):
    id: str
    version: int
    created_by: Optional[str] = None
    created_at: Optional[datetime] = None
    updated_by: Optional[str] = None
    updated_at: Optional[datetime] = None
    parent_template_id: Optional[str] = None
    
    class Config:
        from_attributes = True


class MinutesTemplateList(BaseModel):
    templates: List[MinutesTemplateResponse]
    total: int


# Update GenerateMinutesRequest to include template_id
class GenerateMinutesRequestWithTemplate(BaseModel):
    meeting_id: str
    template_id: Optional[str] = None  # Template ID to use
    include_transcript: bool = True
    format: str = "markdown"  # markdown / html / text

